### PR TITLE
Staff Bros: Balance and Bugfix

### PR DESCRIPTION
**Pull request by cadaeic**

### CODE REVIEW

**Comment by ascriptmaster** [comment]
**Stop closing and opening pull requests, please.**

Also, install Node and run `npm install && npm test` on your command line.

**Comment by cadaeic** [comment]
Sorry for the annoyance caused. :p

**Comment by cadaeic** [comment]
Thanks for the tip.

Currently both custom moves and quotes are solely based on the Pokemon's name, which leads to complications when the name is monkeyed with. Maybe it would be better to add an 'alias' field, for example, to all the mons in order to separate the two.

File: data/scripts.js [hunk 1]
```diff
@@ -3644,9 +3644,20 @@ exports.BattleScripts = {
 		var pool = Object.keys(sets).randomize();
 		var ranks = {'~':'admins', '&':'leaders', '@':'mods', '%':'drivers', '+':'voices'};
 		var levels = {'~':99, '&':97, '@':96, '%':96, '+':95};
-		for (var i = 0; i < 6; i++) {
-			var rank = pool[i].charAt(0);
+		var count = 0;
+		var i = -1;
+		do {
```
Comment (codelegend) [inline data/scripts.js]: why? use a `for` loop.

File: data/scripts.js [hunk 2]
```diff
@@ -3644,9 +3644,20 @@ exports.BattleScripts = {
 		var pool = Object.keys(sets).randomize();
 		var ranks = {'~':'admins', '&':'leaders', '@':'mods', '%':'drivers', '+':'voices'};
 		var levels = {'~':99, '&':97, '@':96, '%':96, '+':95};
-		for (var i = 0; i < 6; i++) {
-			var rank = pool[i].charAt(0);
+		var count = 0;
+		var i = -1;
```
Comment (codelegend) [inline data/scripts.js]: why two variables?? both `i` and `count` have the same value inside the loop body, b/w lines 3651 and 3673

Comment (cadaeic) [inline data/scripts.js]: `i` is the set index, `count` is the number of mons added to the team. If some mons are rejected due to species clause `i` will end up being larger than `count`.

**Comment by Zarel** [comment]
@cadaeic, this is your third pull request on this.

If you make another pull request, I'm going to have to block you for spamming our repository.

It's possible to amend a pull request without making a new one, and people here are very willing to teach you how, if necessary.

**Comment by cadaeic** [comment]
I would very appreciate any help on how to use GitHub better. Thanks in advance and sorry for the annoyance caused.

File: data/scripts.js [hunk 3]
```diff
@@ -3644,9 +3644,20 @@ exports.BattleScripts = {
...
+		do {
+			i++;
 			var set = sets[pool[i]];
+			var rejected = false;
+			for (var otherSet in team) {
+				if (otherSet.species === set.species) {
```
Comment (codelegend) [inline data/scripts.js]: Will this ever be `true`? I don't think so, as there are no duplicate keys.
So, this whole check becomes redundant. (and hence, `count` and `i` are equal).
@Joimer

File: data/scripts.js (hunk 2 above)
Comment (ascriptmaster) [inline data/scripts.js]: We don't actually care about Species Clause for this format

File: config/formats.js [hunk 4]
```diff
@@ -1745,12 +1733,20 @@ exports.Formats = [
 					pokemon.addVolatile('haunterino');
 				};
 			}
-			if (move.id === 'bellydrum' && name === 'jasmine') {
-				move.name = 'Lockdown';
-				move.onHit = function (target, pokemon) {
-					this.add("raw|<div class=\"broadcast-red\"><b>The server is restarting soon.</b><br />Please finish your battles quickly. No new battles can be started until the server resets in a few minutes.</div>");
-				};
-				move.self = {boosts: {atk:6}};
+			if (name === 'jasmine') {
```
Comment (RhythmsAreFromVibrations) [inline config/formats.js]: this worked fine before as ``(move.id === 'bellydrum' && name === 'jasmine')``

when using ``else if``, just make it say ``} else if (move.id === 'transform' && name === 'jasmine') {``

File: data/scripts.js (hunk 2 above)
Comment (cadaeic) [inline data/scripts.js]: @Joimer has given me the go-ahead since having two Gastro is quite frustrating.

File: data/scripts.js (hunk 3 above)
Comment (cadaeic) [inline data/scripts.js]: There are duplicate species (Gastrodon and Blastoise), thus the whole hassle.

**Comment by codelegend** [comment]
squash the commits.
use `git rebase` or `git reset`, and add the changes to a single commit.

**Comment by Zarel** [comment]
Blah, don't close pull requests; you need to learn to amend them instead of making a new one each time you fix something. :|

**Comment by Joimer** [comment]
@cadaeic if you want tomorrow when I'm at work I can teach you how to squash commits and fix a pull request.

**Pull request closed without merging**